menu.py

```diff
@@ -0,0 +1,136 @@
+import datetime
+import re
+
+from file_system import FileManipulator
+from spending import Spending
+from category import Category
+
+from matplotlib import pyplot as plt  # type: ignore
+
+
+class Menu:
+    def show_menu(self) -> None:
+        print("Enter filename: ")
+        filename = str(input())
+        if ".csv" not in filename:
+            filename += ".csv"
+        file = FileManipulator(filename=filename)
+
+        while True:
+            print("1. Add new spending")
+            print("2. Show all spending")
+            print("3. Show spending by date")
+            print("4. Show spending by category")
+            print("5. Show diagram")
+            print("6. Add new category")
+            print("7. Exit")
+
+            option = int(input("Enter number: "))
+
+            if option == 1:
+                self.add_new_category(file)
+            elif option == 2:
+                self.show_all_spendings(file)
+            elif option == 3:
+                self.show_spending_by_date(file)
+            elif option == 4:
+                self.show_spending_by_category(file)
+            elif option == 5:
+                self.show_diagram(file)
+            elif option == 6:
+                self.add_new_category()
+            elif option == 7:
+                self.exit(0)
+            else:
+                raise Exception(ValueError("invalid value."))
+
+    def add_new_spending(self, file: FileManipulator) -> None:
```

Comment:
if u pass filename as constructor parameter, then u can just pass self and define file_manipulator in __init__ method